openstack_virtual_baremetal/deploy.py

```diff
@@ -354,11 +356,14 @@ def _process_role(role_file, base_envs, stack_name, args):
     _add_identifier(role_env, 'overcloud_storage_net', args.id)
     _add_identifier(role_env, 'overcloud_storage_mgmt_net', args.id)
     _add_identifier(role_env, 'overcloud_tenant_net', args.id)
-    role_env['parameter_defaults']['networks'] = {
-        'private': role_env['parameter_defaults']['private_net'],
-        'provision': role_env['parameter_defaults']['provision_net'],
-        'public': role_env['parameter_defaults']['public_net'],
-        }
+    if not role_env['parameter_defaults'].get('networks'):
+        role_env['parameter_defaults']['networks'] = {}
+    for k, v in {'private': 'private_net',
+                 'provision': 'provision_net',
+                 'public': 'public_net'}.items():
+        role_env['parameter_defaults']['networks'].update(
```

Comment:
Is this to maintain compatibility with network names being passed as parameters? Because for new features we shouldn't need to do that. Anything that gets added after the switch to "everything in parameter_defaults" was made only needs to respect parameter_defaults.

Reply:
A unit test was failing with key error.

$ `git checkout 096f7217d6883927eac3a6a7bbdde9d32418064e`
$ `tox -e py27`

>   FAIL: tests.test_deploy.TestDeploy.test_process_role
> tags: worker-0
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/tmp/openstack-virtual-baremetal/.tox/py27/lib/python2.7/site-packages/mock/mock.py", line 1305, in patched
>     return func(*args, **keywargs)
>   File "/tmp/openstack-virtual-baremetal/openstack_virtual_baremetal/tests/test_deploy.py", line 389, in test_process_role
>     'foo', args)
>   File "openstack_virtual_baremetal/deploy.py", line 334, in _process_role
>     'private': role_env['parameter_defaults']['private_net'],
> KeyError: 'private_net'

Reply:
Foo, that one's my fault. Since we aren't going to be able to support those being passed as parameters anyway, we should probably just move the parameters to parameter_defaults in the unit test. There's no point testing a case that won't work anyway.

Reply:
Ok, I reverted the change in deploy.py and updated the unit test to include the parameters in parameters_defaults. commit 026eeb342b7ff17a0a2f5cd9ee639fc1499a671b